src/ui/public/system_api/system_api.js

```diff
@@ -0,0 +1,31 @@
+const SYSTEM_API_HEADER_NAME = 'kbn-system-api';
+
+export default function SystemApisProvider() {
+
+  /**
+   * Adds a custom header designating request as system API
+   * @param headers Object representing set of headers
+   * @return Object representing set of headers, with system API header added in
+   */
+  function addSystemApiHeader(headers) {
+    const newHeaders = headers;
+    newHeaders[SYSTEM_API_HEADER_NAME] = true;
+    return newHeaders;
```

Comment:
That's not a copy, but just a new reference. I think @epixa meant something akin to:

```js
return Object.assign(headers, { [SYSTEM_API_HEADER_NAME]: true });
```

Or more ES6y:

```js
return {
  ...headers,
  [SYSTEM_API_HEADER_NAME]: true
};
```